WcaOnRails/app/models/abstract_cached_model.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+class AbstractCachedModel < ActiveRecord::Base
```

Comment:
This class is very hard to understand. Maybe because of the complexity of handling `@@` variables in a base class. It's probably clearer to just implement things directly in each class.

Redefining a core framework method like `find` scares me. I don't know who might be calling it and expecting the regular functionality.

We also need to figure out if and how to deal with cache invalidation.